components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/store/UserSessionStore.java

```diff
@@ -685,4 +686,47 @@ public boolean isExistingMapping(User user, int idpId, String sessionId) throws
         }
         return isExisting;
     }
+
+    /**
+     * Store session details of a given session context key to map the session context key with
+     * the federated IdP's session ID.
+     *
+     * @param sessionContextKey Session Context Key.
+     * @param authHistory       History of the authentication flow.
+     * @throws UserSessionException Error while storing session details.
+     */
+    public void storeFederatedAuthSessionInfo(String sessionContextKey, AuthHistory authHistory)
+            throws UserSessionException {
+
+        try (Connection connection = IdentityDatabaseUtil.getDBConnection(false);
+             PreparedStatement prepStmt = connection.prepareStatement(SQLQueries.SQL_STORE_FEDERATED_AUTH_SESSION_INFO)) {
+            prepStmt.setString(1, authHistory.getIdpSessionIndex());
+            prepStmt.setString(2, sessionContextKey);
+            prepStmt.setString(3, authHistory.getIdpName());
+            prepStmt.setString(4, authHistory.getAuthenticatorName());
+            prepStmt.setString(5, authHistory.getRequestType());
+            prepStmt.execute();
+        } catch (SQLException e) {
```

Comment:
Don't we need a rollback here?

Reply:
Fixed with PR https://github.com/wso2/carbon-identity-framework/pull/3738